### PR TITLE
Refactor of misc entity functions

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Little bit of cleanup after the split pr

### CODE REVIEW

**Review by IntelOrca — APPROVED** [review]

File: src/openrct2/peep/Peep.cpp [hunk 1]
```diff
@@ -474,7 +474,9 @@ std::optional<CoordsXY> Peep::UpdateAction(int16_t& xy_distance)
     WindowInvalidateFlags |= PEEP_INVALIDATE_PEEP_2;
 
     // Create sick at location
-    litter_create({ x, y, z, sprite_direction }, (sprite_index & 1) ? LITTER_TYPE_SICK_ALT : LITTER_TYPE_SICK);
+    Litter::Create(
+        { x, y, z, sprite_direction },
+        (sprite_index & 1) ? Litter::Type::SickAlt : Litter::Type::Sick);
```
Comment (tupaschoal) [inline src/openrct2/peep/Peep.cpp:479]:
```suggestion
    Litter::Create({ x, y, z, sprite_direction }, (sprite_index & 1) ? Litter::Type::SickAlt : Litter::Type::Sick);
```

File: src/openrct2/peep/Peep.cpp [hunk 2]
```diff
@@ -1954,7 +1956,8 @@ static void peep_footpath_move_forward(Peep* peep, const CoordsXYE& coords, bool
                 continue;
 
             litter_count++;
-            if (litter->SubType != LITTER_TYPE_SICK && litter->SubType != LITTER_TYPE_SICK_ALT)
+            if (litter->SubType != Litter::Type::Sick
+                && litter->SubType != Litter::Type::SickAlt)
```
Comment (tupaschoal) [inline src/openrct2/peep/Peep.cpp:1960]:
```suggestion
            if (litter->SubType != Litter::Type::Sick && litter->SubType != Litter::Type::SickAlt)
```

**Review by tupaschoal — APPROVED** [review]
This looks fine. I've fixed the formatting on a 5th commit, ideally it should be folded into the 4th.

This will also have merge conflicts with #14642 (or vice-versa)

File: src/openrct2/world/Litter.h [hunk 3]
```diff
@@ -15,31 +15,29 @@ class DataSerialiser;
...
-    LITTER_TYPE_EMPTY_DRINK_CARTON,
-    LITTER_TYPE_EMPTY_JUICE_CUP,
-    LITTER_TYPE_EMPTY_BOWL_BLUE,
-};
-
 struct Litter : SpriteBase
 {
+    enum class Type : uint8_t
+    {
+        Sick,
```
Comment (Gymnasiast) [inline src/openrct2/world/Litter.h:22]: Let's change this to `Vomit` - "Sick" is a colloquialism.

File: src/openrct2/world/Sprite.cpp [hunk 4]
```diff
@@ -763,6 +750,25 @@ void litter_remove_at(const CoordsXYZ& litterPos)
     }
 }
 
+const rct_string_id litterNames[12] = { STR_LITTER_VOMIT,
```
Comment (Gymnasiast) [inline src/openrct2/world/Sprite.cpp:753]: Can we make this static?

**Pull request merged**